label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -243,9 +244,10 @@ public void validateLabelExpression(Action.DynamicLabel dynamicLabel) {
         .getTokenExtractionRulesList()
         .forEach(
             tokenExtractionRule -> {
-              validKeys.add(tokenExtractionRule.getKey());
               if (tokenExtractionRule.hasAlias()) {
                 validKeys.add(tokenExtractionRule.getAlias());
+              } else {
```

Comment:
Unintentional change?

Reply:
Nope. If alias is provided then alias should be used not key.

Reply:
Okay. So, till now we weren't using alias at all?

Reply:
Earlier both alias and key were being used. I will change to use alias if alias is provided.
Please note this expression is not exposed in UI so far, so no one should be using it.